File: platform/fivem-adapter.js
// Bucu Core - FiveM Adapter (JavaScript)
// Bridges Core and FiveM-specific APIs

const FiveMAdapter = {
    _initialized: false,
    
    // Initialize FiveM adapter
    init() {
        if (this._initialized) {
            if (typeof Logger !== 'undefined') {
                Logger.warn("FiveM Adapter already initialized");
            }
            return false;
        }
        
        if (typeof Logger !== 'undefined') {
            Logger.info("Initializing FiveM Adapter (JS)");
        }
        
        // Note: Player connection/disconnection is handled by Lua side
        // JS side focuses on event bridging and utility functions
        
        // Register network events for cross-language communication
        if (typeof RegisterNetEvent !== 'undefined') {
            RegisterNetEvent("bucu:clientEventJS");
            on("bucu:clientEventJS", (eventName, data) => {
                const source = global.source;
                
                if (typeof EventSystem !== 'undefined') {
                    EventSystem.emit(`client:${eventName}`, {
                        source: source,
                        data: data
                    });
                }
            });
        }
        
        this._initialized = true;
        
        if (typeof Logger !== 'undefined') {
            Logger.info("FiveM Adapter (JS) initialized");
        }
        
        return true;
    },
    
    // Trigger client event
    triggerClient(source, eventName, data) {
        if (typeof source !== "number") {
            if (typeof Logger !== 'undefined') {
                Logger.error("Source must be a number");
            }
            return false;
        }
        
        if (typeof eventName !== "string" || eventName === "") {
            if (typeof Logger !== 'undefined') {
                Logger.error("Event name must be a non-empty string");
            }
            return false;
        }
        
        if (typeof TriggerClientEvent !== 'undefined') {
            TriggerClientEvent(eventName, source.toString(), data);
        }
        return true;
    },
    
    // Trigger client event for all players
    triggerClientAll(eventName, data) {
        if (typeof eventName !== "string" || eventName === "") {
            if (typeof Logger !== 'undefined') {
                Logger.error("Event name must be a non-empty string");
            }
            return false;
        }
        
        if (typeof TriggerClientEvent !== 'undefined') {
            TriggerClientEvent(eventName, "-1", data);
        }
        return true;
    },
    
    // Get player name
    getPlayerName(source) {
        if (typeof GetPlayerName !== 'undefined') {
            return GetPlayerName(source.toString());
        }
        return null;
    },
    
    // Get player identifiers
    getPlayerIdentifiers(source) {
        if (typeof GetPlayerIdentifiers !== 'undefined') {
            return GetPlayerIdentifiers(source.toString());
        }
        return [];
    },
    
    // Get player ping
    getPlayerPing(source) {
        if (typeof GetPlayerPing !== 'undefined') {
            return GetPlayerPing(source.toString());
        }
        return 0;
    },
    
    // Get player endpoint (IP)
    getPlayerEndpoint(source) {
        if (typeof GetPlayerEndpoint !== 'undefined') {
            return GetPlayerEndpoint(source.toString());
        }
        return null;
    },
    
    // Get all player sources
    getPlayers() {
        if (typeof GetPlayers !== 'undefined') {
            const players = GetPlayers();
            return players.map(p => parseInt(p));
        }
        return [];
    },
    
    // Check if player exists
    playerExists(source) {
        return this.getPlayerName(source) !== null;
    },
    
    // Drop player (kick)
    dropPlayer(source, reason = "Kicked") {
        if (typeof DropPlayer !== 'undefined') {
            DropPlayer(source.toString(), reason);
        }
        return true;
    },
    
    // Register server event
    registerServerEvent(eventName) {
        if (typeof RegisterNetEvent !== 'undefined') {
            RegisterNetEvent(eventName);
        }
        return true;
    },
    
    // Trigger server event
    triggerServerEvent(eventName, ...args) {
        if (typeof TriggerEvent !== 'undefined') {
            TriggerEvent(eventName, ...args);
        }
        return true;
    }
};

// Export for Node.js and FiveM
if (typeof module !== 'undefined' && module.exports) {
    module.exports = FiveMAdapter;
}

// Make available globally in FiveM
if (typeof global !== 'undefined') {
    global.FiveMAdapter = FiveMAdapter;
}

File: platform/fivem-adapter.lua
-- Bucu Core - FiveM Adapter
-- Bridges Core and FiveM-specific APIs

local FiveMAdapter = {
    _initialized = false
}

-- Initialize FiveM adapter
function FiveMAdapter:init()
    if self._initialized then
        if Logger then
            Logger:warn("FiveM Adapter already initialized")
        end
        return false
    end
    
    if Logger then
        Logger:info("Initializing FiveM Adapter")
    end
    
    -- Register player connection handler
    AddEventHandler("playerConnecting", function(name, setKickReason, deferrals)
        local source = source
        
        deferrals.defer()
        
        -- Wait a bit for player data to be available
        Wait(0)
        
        -- Get player identifiers
        local identifiers = GetPlayerIdentifiers(source)
        local identifier = identifiers[1] or "unknown"
        
        -- Create player object
        if PlayerManager then
            PlayerManager:createPlayer(source, name, identifier)
        end
        
        deferrals.done()
    end)
    
    -- Register player dropped handler
    AddEventHandler("playerDropped", function(reason)
        local source = source
        
        if Logger then
            Logger:debug(string.format("Player %d dropped: %s", source, reason))
        end
        
        -- Remove player object
        if PlayerManager then
            PlayerManager:removePlayer(source)
        end
    end)
    
    -- Register resource start handler
    AddEventHandler("onResourceStart", function(resourceName)
        if GetCurrentResourceName() == resourceName then
            if Logger then
                Logger:info("Bucu Core started")
            end
            
            -- Emit core ready event
            if EventSystem then
                EventSystem:emit("core:ready", {})
            end
        end
    end)
    
    -- Register resource stop handler
    AddEventHandler("onResourceStop", function(resourceName)
        if GetCurrentResourceName() == resourceName then
            if Logger then
                Logger:info("Bucu Core stopping")
            end
            
            -- Emit core shutdown event
            if EventSystem then
                EventSystem:emit("core:shutdown", {})
            end
        end
    end)
    
    -- Register network events for cross-language communication
    RegisterNetEvent("bucu:clientEvent")
    AddEventHandler("bucu:clientEvent", function(eventName, data)
        local source = source
        
        if EventSystem then
            EventSystem:emit("client:" .. eventName, {
                source = source,
                data = data
            })
        end
    end)
    
    self._initialized = true
    
    if Logger then
        Logger:info("FiveM Adapter initialized")
    end
    
    return true
end

-- Trigger client event
function FiveMAdapter:triggerClient(source, eventName, data)
    if type(source) ~= "number" then
        if Logger then
            Logger:error("Source must be a number")
        end
        return false
    end
    
    if type(eventName) ~= "string" or eventName == "" then
        if Logger then
            Logger:error("Event name must be a non-empty string")
        end
        return false
    end
    
    TriggerClientEvent(eventName, source, data)
    return true
end

-- Trigger client event for all players
function FiveMAdapter:triggerClientAll(eventName, data)
    if type(eventName) ~= "string" or eventName == "" then
        if Logger then
            Logger:error("Event name must be a non-empty string")
        end
        return false
    end
    
    TriggerClientEvent(eventName, -1, data)
    return true
end

-- Get player name
function FiveMAdapter:getPlayerName(source)
    return GetPlayerName(source)
end

-- Get player identifiers
function FiveMAdapter:getPlayerIdentifiers(source)
    return GetPlayerIdentifiers(source)
end

-- Get player ping
function FiveMAdapter:getPlayerPing(source)
    return GetPlayerPing(source)
end

-- Get player endpoint (IP)
function FiveMAdapter:getPlayerEndpoint(source)
    return GetPlayerEndpoint(source)
end

-- Get all player sources
function FiveMAdapter:getPlayers()
    return GetPlayers()
end

-- Check if player exists
function FiveMAdapter:playerExists(source)
    return GetPlayerName(source) ~= nil
end

-- Drop player (kick)
function FiveMAdapter:dropPlayer(source, reason)
    DropPlayer(source, reason or "Kicked")
    return true
end

-- Register server event
function FiveMAdapter:registerServerEvent(eventName)
    RegisterNetEvent(eventName)
    return true
end

-- Trigger server event
function FiveMAdapter:triggerServerEvent(eventName, ...)
    TriggerEvent(eventName, ...)
    return true
end

return FiveMAdapter
